pkg/services/ngalert/state/state_test.go

```diff
@@ -73,3 +75,93 @@ func TestNeedsSending(t *testing.T) {
 		})
 	}
 }
+
+func TestSetEndsAt(t *testing.T) {
+	evaluationTime, _ := time.Parse("2006-01-02", "2021-03-25")
+	testCases := []struct {
+		name       string
+		expected   time.Time
+		testState  *State
+		testRule   *ngmodels.AlertRule
+		testResult eval.Result
+	}{
+		{
+			name:      "For: unset Interval: 10s",
+			expected:  evaluationTime.Add(20 * time.Second),
+			testState: &State{},
+			testRule: &ngmodels.AlertRule{
+				IntervalSeconds: 10,
+			},
+			testResult: eval.Result{
+				EvaluatedAt: evaluationTime,
+			},
+		},
+		{
+			name:      "For: 0s Interval: 10s",
```

Comment:
nitpick, not merge blocking.

But might be easier for people to follow tests the names have a "should clause" that describes the behavior. e.g. "EndsAt should be twice the interval if FOR is 0." - so worded a bit more spec-like.

Reply:
Easy enough to add